### ISSUE TITLE
Does this component support grouping?

### ISSUE THREAD

**Issue by bunomonteiro** [KazanExpress/vue-simple-suggest#59]
Does this component support grouping (similar to optgroup)? 
I didn't see in the doc.

The doc tells us about 'custom item', but I need to customize one level up (group of items)

**Comment by kaskar2008** [comment]
Hello @bunomonteiro.
Unfortunately this component doesn't support grouping yet, but this is quite interesting feature, so we will implement it soon.
@raiondesu, need your opinion.
@bunomonteiro, feel free to contribute :)

**Comment by Raiondesu** [comment]
Yes, grouping is not technically supported currently, but it would be nice to have some way of extending the component's template to make such things possible.

This issue is definitely a next minor version material.

**Comment by Raiondesu** [comment]
@bunomonteiro, on the second thought, you can currently achieve this with a following workaround:
Define the splitters for your groups in suggestions list:
```js
[
  {
    id: 1,
    title: 'Group #1',
    splitter: true
  },
  {
    id: 2,
    title: 'some item from group 1'
  },
  {
    id: 3,
    title: 'another item from group 1'
  },
  {
    id: 4,
    title: 'Group #2 - the fancy one',
    splitter: true
  },
  {
    id: 5,
    title: 'WOOW, some item from group 2'
  },
  {
    id: 6,
    title: 'another item from group 2'
  }
]
```

And then, in your suggestion slot template you could define something like this:

```html
<div slot="suggestion-item" slot-scope="{ suggestion }" :title="suggestion.title">
  <span v-if="suggestion.splitter" @click.stop="null">{{ suggestion.title }}</span>
  <span v-else>{{ suggestion.title }}</span>
</div>
```

This can help with **imitating** suggestions grouping.

The official documented support is coming in the next minor version.

**Comment by bunomonteiro** [comment]
I got it!

Because of component (today) limitations it is necessary to use an array of flat objects. I see that.
But it would be more interesting if the component supported an array of complex objects. Of course, in the next version.
Like this:

```javascript
[
  {
    _id: "as78d5as87das6",
    title: "Some Group Title",
    children: [
      {
        _id: "098asdf09asd78f",
        title: "Some child's title",
        value: "some_value",
      }, { ... }
    ]
  }, { ... } 
]
```

**Comment by Raiondesu** [comment]
@bunomonteiro, I'd like to correct your words a bit: not __array of flat objects__, but rather __flat array of objects__, because it's possible to provide dotted paths to `display-attribute` and `value-attribute` props:
`display-attribute="info.general.title"`

About defining future formats to support nesting...
I though of something like a `nesting-attribute`. It would be possible to achieve nesting for your example like this:
```html
<vue-simple-suggest ref="vueSimpleSuggest" v-model="model"
  value-attribute="id"
  display-attribute="title"
  nesting-attribute="children"
></vue-simple-suggest>
```

@kaskar2008, what do you think?

**Comment by bunomonteiro** [comment]
hum...
Isn't an **array of flat objects**, but at the end works like it.

About your proposal for future format, sounded great to me.

``` html
<vue-simple-suggest ref="vueSimpleSuggest" v-model="model"
  value-attribute="id"
  display-attribute="title"
  children-attribute="children"
>
<div slot="suggestion-item">...</div>
<div slot="suggestion-child">...</div>
</vue-simple-suggest>
```

**Comment by Raiondesu** [comment]
@bunomonteiro, we rememeber.
:)

**Comment by wojewsky** [comment]
Grouping would be a great feature.
Do you have a release plan for it?

**Comment by kaskar2008** [comment]
@wojewsky Yes, ofc we have, but we still can't find a good and flexible solution for this case. Feel free to help us with it :)

**Comment by wojewsky** [comment]
in my opinion the array is a very good way to group some values, but it should be possible to define nonselectable groupheadlines (eg by a flag 'notSelectable')

**Comment by bunomonteiro** [comment]
> @bunomonteiro, on the second thought, you can currently achieve this with a following workaround:
> Define the splitters for your groups in suggestions list:
> 
> ```js
> [
>   {
>     id: 1,
>     title: 'Group #1',
>     splitter: true // This flag will allow us to determine that the current suggestion is a group splitter
>   },
>   {
>     id: 2,
>     title: 'some item from group 1'
>   },
>   {
>     id: 3,
>     title: 'another item from group 1'
>   },
>   {
>     id: 4,
>     title: 'Group #2 - the fancy one',
>     splitter: true
>   },
>   {
>     id: 5,
>     title: 'WOOW, some item from group 2'
>   },
>   {
>     id: 6,
>     title: 'another item from group 2'
>   }
> ]
> ```
> 
> And then, in your suggestion slot template you could define something like this:
> 
> ```
> <div slot="suggestion-item" slot-scope="{ suggestion }" :title="suggestion.title">
>   
>   <span v-if="suggestion.splitter"
>     @click.stop="null"
>     @keyup.stop="null"
>     @mouseenter.stop="null"
>     @mouseleave.stop="null"
>     @keydown.stop="null"
>     class="your-custom-splitter-class"
>   >{{ suggestion.title }}</span>
>   
>   <span v-else>{{ suggestion.title }}</span>
> </div>
> ```
> 
> This very hacky solution can help with **imitating** suggestions grouping.
> 
> The official documented support is coming in the next minor (or maybe major) version.

I used this snippet right now and it worked perfectly! :clap::clap:
